### DISCUSSION TITLE
Attribute Limit Exceeded

### DISCUSSION THREAD

**Discussion by JosephShering** [appwrite/appwrite#4562] [Q&A] [answered]
### 👟 Reproduction steps

1. Create a collection
2. Start adding URL attributes, after maybe 4-5 you will get a popup error saying, "Attribute Limit Exceeded"

### 👍 Expected behavior

Should be able to create as many attributes as I want, or at least very many before this limit is reached.

### 👎 Actual Behavior

Can only make a 2-4 URL attributes on a document before getting an error.

<img width="994" alt="image" src="https://user-images.githubusercontent.com/5262224/174737273-981f02cf-4e9e-45f2-92cd-68284e9f900c.png">

### 🎲 Appwrite version

Version 0.14.x

### 💻 Operating system

Linux

### 🧱 Your Environment

I am using the digital ocean marketplace version of Appwrite

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/appwrite/appwrite/blob/HEAD/CODE_OF_CONDUCT.md)

**Comment by Meldiron** [comment, marked as answer]
Hey there 👋 Our URL attribute has a size set to `2000`, and that counts towards the 16kb attribute limit (defined by MariaDB).

That means you should be able to add up to 8 URL attributes if you had no other attributes.

In a screenshot I can see a string attribute, what size did you set for that? It also counts towards this 16kb limit and might be using most of it.

There is a breaking point for string sizes, and it is `16384`. If you set the size of your string attribute to the number above that, it will NOT count towards the 16k limit and you will have much more space for other attributes. The only problem tho will be indexing - it is not as fast as it would be if you defined the number below it.

**Reply by ZeroAurora** [reply]
> There is a breaking point for string sizes, and it is `16384`. If you set the size of your string attribute to the number above that, it will NOT count towards the 16k limit and you will have much more space for other attributes. The only problem tho will be indexing - it is not as fast as it would be if you defined the number below it.

I spent some time debugging this and come across this discussion. Is it possible to have this kind of string attribute directly called "large text" or something? Or can we make the error more specific or documented so that others won't waste time debugging?

**Comment by stnguyen90** [comment]
@JosephShering, do you have any other questions about this or can this be closed?
